File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        int ans = -1;
        string first = strs[0];
        int n = strs.size();
        string last = strs[n-1];

        int m = min(first.length(),last.length());
        for(int i = 0 ; i <m;i++){
            if(first[i]!=last[i]){
                break;
            }
            else{
                ans++;
            }
            
        }

        string a = "";
        for(int i = 0 ; i <=ans ; i++){
            a.push_back(first[i]);
        }
        return a;
        
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        strs.sort()
        ans = ''
        a = strs[0]
        b = strs[-1]
        for i in range(len(a)):
            if a[i]==b[i]:
                ans+=a[i]
            else:
                break
        return ans
        
        